waspc/src/Wasp/ConfigFile.hs

```diff
@@ -0,0 +1,52 @@
+module Wasp.ConfigFile
+  ( discoverConfigFiles,
+    tailwindConfigFile,
+    postcssConfigFile,
+    ConfigFileRelocator (..),
+  )
+where
+
+import Data.Map (fromList)
+import qualified Data.Map as Data
+import Data.Maybe (mapMaybe)
+import StrongPath (Abs, Dir, File', Path', Rel, castRel, relfile, (</>))
+import Wasp.Common (WaspProjectDir)
+import Wasp.Generator.Common (ProjectRootDir)
+import Wasp.Generator.WebAppGenerator.Common (webAppRootDirInProjectRootDir)
+import qualified Wasp.Util.IO as Util.IO
+
+-- | A type for establishing the mapping of where to copy config files from/to.
+data ConfigFileRelocator = ConfigFileRelocator
+  { _pathInWaspDir :: Path' Abs File',
+    _projectRootDirPath :: Path' (Rel ProjectRootDir) File'
```

Comment:
Changed to `pathInProjectRootDir` for consistency, thanks for noting. 👍🏻 

I don't think I can change the type of `pathInWaspDir` to relative, since I need to construct a `createCopyFileDraft :: Path' (Rel ProjectRootDir) File' -> Path' Abs File' -> FileDraft` during generation, where my signature is `genConfigFiles :: AppSpec -> Generator [FileDraft]` and waspDir path is long gone by then. I don't think we do this in other places that need that info either, as AppSpec has other abs file paths. We could, in theory, inject waspDir into the AppSpec so others can use it and we can keep it all relative, but that feels like a pretty big shift to make atm.

Reply:
Hah while reading at first glance I thought you wrote "thanks for nothing" (while you wrote "thanks for noting") and was like "woah, where did that come from" :D.

Ok, got it regarding pathInWaspDir -> if that is so, then it makes sense.
One smaller way to do it would be to insert it right here, although I see why then it feels like it is better to insert it for the whole AppSpec which then requires bigger refactoring. Sounds good, we can leave it as it is for now, it will be fine in practice, if not we can refactor later.

Reply:
Haha, oh man, "thanks for nothing" would indicate a really bad day :D Yep, let's do the add waspDir to AppSpec thing like we both were thinking so we can stay relative here and really solve that other problem! 👍🏻

Reply:
As long as it is not "Yeah, thanks for another stupid comment Martin, I never have enough of those" I am good :D. And even than I am sure it will be deserved :D.